### PR TITLE
DSCResources: Update Pester syntax to v4

### LINKED ISSUE #327: Migrate Pester test syntax from v3 -> v4 in test templates
As of Pester v4 the syntax of the Should cmdlet has changed from Should Be to Should -Be.

There is a helpful script that can be used to easily perform the conversion:
https://gist.github.com/indented-automation/aeb14825e39dd8849beee44f681fbab3

This script should be used to correct the Pester syntax.

### PR DESCRIPTION
**Pull request by mnewswanger**
Resolves #327

---
This change is [Reviewable](https://reviewable.io/reviews/powershell/dscresources/329)

### CODE REVIEW

**Comment by PlagueHO** [comment]
Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[Tests.Template/integration_template.ps1, line 58 at r1](https://reviewable.io:443/reviews/powershell/dscresources/329#-Kwi7g0-yLXsfvooFc_w:-Kwi7g0-yLXsfvooFc_x:b-c4gweu) ([raw file](https://github.com/powershell/dscresources/blob/55d1eee084910d2daf928b896d2b73b0f91c40d6/Tests.Template/integration_template.ps1#L58)):*
> ```PowerShell
>         It 'Should be able to call Get-DscConfiguration without throwing' {
>             {
> 				Get-DscConfiguration -Verbose -ErrorAction Stop 
> ```

Could you convert the indent on this line and the next to spaces?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/dscresources/329)*

**Comment by mnewswanger** [comment]
Review status: 1 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[Tests.Template/integration_template.ps1, line 58 at r1](https://reviewable.io:443/reviews/powershell/dscresources/329#-Kwi7g0-yLXsfvooFc_w:-Kwjmx9zcTQdMhShvTk8:bb6un1j) ([raw file](https://github.com/powershell/dscresources/blob/55d1eee084910d2daf928b896d2b73b0f91c40d6/Tests.Template/integration_template.ps1#L58)):*
<details><summary><i>Previously, PlagueHO (Daniel Scott-Raynsford) wrote…</i></summary><blockquote>

Could you convert the indent on this line and the next to spaces?
</blockquote></details>

Done

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/dscresources/329)*

**Comment by PlagueHO** [comment]
:lgtm:

Thanks @mnewswanger

---

Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/dscresources/329#-:-KwsXYwUIjJ38FGrhEEA:b-683ep6)*

**Comment by PlagueHO** [comment]
@KwirkyKat , are you able to merge if you're happy?

---

Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/dscresources/329#-:-KwsXn_9UV3VZbGoW8Km:bqg5r55)*

File: Tests.Template/changelist.md [hunk 1]
```diff
@@ -21,6 +21,9 @@ When the templates are used to create or update tests in a DSC Resource the vers
 * First release including version number.
 
 ## unit_template.ps1
+### Unreleased ###
+* Update Pester syntax to v4
+
```
Comment (kwirkykat) [inline Tests.Template/changelist.md]: These test templates don't get released. They just stay here in this repo for other people to copy.
So can you increment the version number in the comment in the template and put the same version number here in place of 'Unreleased' as well?

**Pull request merged**